Tests/Unit/ExtensionTest.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Trikoder\Bundle\OAuth2Bundle\Tests\Unit;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Trikoder\Bundle\OAuth2Bundle\DependencyInjection\TrikoderOAuth2Extension;
+use Trikoder\Bundle\OAuth2Bundle\Manager\InMemory\ScopeManager;
+use Trikoder\Bundle\OAuth2Bundle\Manager\ScopeManagerInterface;
+
+final class ExtensionTest extends TestCase
+{
+    /**
+     * @dataProvider grantsProvider
+     */
+    public function testEnablingAndDisablingGrants(string $referenceId, string $grantKey, bool $shouldTheGrantBeEnabled): void
+    {
+        $container = new ContainerBuilder();
+
+        $this->setupContainer($container);
+
+        $extension = new TrikoderOAuth2Extension();
+
+        $extension->load($this->getValidConfiguration([$grantKey => $shouldTheGrantBeEnabled]), $container);
```

Comment:
@X-Coder264 You could always do an `array_merge` instead of passing `$options`, but I don't know whether that's better or worse tbh.

Reply:
@HypeMC I was thinking of doing that originally, but this seems cleaner to me for now (as the array is more than one level deep).